src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -0,0 +1,179 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using MSBuild = Microsoft.Build.Utilities;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading;
+using System.IO;
+using Microsoft.DotNet.Build.CloudTestTasks;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Build.Tasks.Feed
+{
+    sealed class BlobFeedAction
+    {
+        private MSBuild.TaskLoggingHelper Log;
+        private static readonly CancellationTokenSource TokenSource = new CancellationTokenSource();
+        private static readonly CancellationToken CancellationToken = TokenSource.Token;
+        private static readonly string IndexDirectory = Path.Combine(Directory.GetCurrentDirectory(), "tmp");
+
+        public BlobFeed feed;
+        public int MaxClients { get; set; } = 8;
+
+        public BlobFeedAction(string accountName, string accountKey, string containerName, MSBuild.TaskLoggingHelper Log)
+        {
+            this.feed = new BlobFeed(accountName, accountKey, containerName, Log);
+            this.Log = Log;
+        }
+
+        public async Task<bool> PushToFeed(IEnumerable<string> items, string relativePath, bool containerExists = false)
+        {
+            if (feed.IsSanityChecked(items))
+            {
+                if (CancellationToken.IsCancellationRequested)
+                {
+                    Log.LogError("Task PushToFeed cancelled");
+                    CancellationToken.ThrowIfCancellationRequested();
+                }
+
+                List<string> indexList = new List<string>(Directory.EnumerateFiles(Path.Combine(IndexDirectory, relativePath), "index.json", SearchOption.AllDirectories));
+                List<string> listAzureBlobs = await ListAzureBlobs.ListBlobs(Log, feed.AccountName, feed.AccountKey, feed.ContainerName, string.Empty);
+                using (var clientThrottle = new SemaphoreSlim(this.MaxClients, this.MaxClients))
+                {
+                    Func<string, string> calculatePath = (string item) =>
+                    {
+                        Tuple<string, NuGetVersion> blobPath = feed.CalculateBlobPath(item);
+                        string relativeBlobPath = Path.Combine(relativePath, blobPath.Item1, blobPath.Item2.ToFullString());
+                        return Path.Combine(relativeBlobPath, Path.GetFileName(item)).ToLowerInvariant().Replace("\\", "/");
+                    };
+                    await Task.WhenAll(items.Select(item => UploadAsync(CancellationToken, item, calculatePath(item), clientThrottle, listAzureBlobs)));
+
+                    if (containerExists)
+                    {
+                        listAzureBlobs = await ListAzureBlobs.ListBlobs(Log, feed.AccountName, feed.AccountKey, feed.ContainerName, string.Empty);
+                        items.All(item => UpdateIndexes(item, relativePath, listAzureBlobs));
```

Comment:
I would probably just add your new packages to the list instead of listing them from azure again.